src/main/java/mate/academy/intro/mapper/BookMapper.java

```diff
@@ -37,8 +38,8 @@ default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
     }
 
     @AfterMapping
-    default void setCategories(CreateBookRequestDto requestDto,@MappingTarget Book book) {
-        CategoryRepository categoryRepository = ContextLoader.getCurrentWebApplicationContext()
+    default void setCategories(CreateBookRequestDto requestDto, @MappingTarget Book book) {
+        CategoryRepository categoryRepository = ApplicationContextUtil.getApplicationContext()
```

Comment:
you actually do not need a repo here
you can create new Category() and set only id field to it))